pilot/pkg/model/context.go

```diff
@@ -643,14 +643,14 @@ func (node *Proxy) GetNetworkView() map[string]bool {
 
 // InNetwork returns true if the proxy is on the given network, or if either
 // the proxy's network or the given network is unspecified ("").
-func (node *Proxy) InNetwork(network string) bool {
-	return node == nil || SameOrEmpty(network, node.Metadata.Network)
+func (node *Proxy) InNetwork(network NetworkID) bool {
+	return node == nil || SameOrEmpty(string(network), string(node.Metadata.Network))
 }
 
 // InCluster returns true if the proxy is in the given cluster, or if either
 // the proxy's cluster id or the given cluster id is unspecified ("").
-func (node *Proxy) InCluster(cluster string) bool {
-	return node == nil || SameOrEmpty(cluster, node.Metadata.ClusterID)
+func (node *Proxy) InCluster(cluster ClusterID) bool {
+	return node == nil || SameOrEmpty(string(cluster), string(node.Metadata.ClusterID))
```

Comment:
Consider defining a String() for the type ClusterID to have a better control of how the type ClusterID is converted to string.